metrics/metrics.go

```diff
@@ -0,0 +1,37 @@
+// Package metrics provides an extensible framework to instrument your
+// application. All metrics are safe for concurrent use. Considerable design
+// influence has been taken from https://github.com/codahale/metrics and
+// https://prometheus.io.
+package metrics
+
+// Counter is a monotonically-increasing, unsigned, 64-bit integer used to
+// capture the number of times an event has occurred. By tracking the deltas
+// between measurements of a counter over intervals of time, an aggregation
+// layer can derive rates, acceleration, etc.
+type Counter interface {
```

Comment:
The name is really only relevant at export-time, and so can be built in to the construction process. The "name" at observation time is the variable name.